rest_framework/generics.py

```diff
@@ -18,7 +19,7 @@ def get_object_or_404(queryset, *filter_args, **filter_kwargs):
     """
     try:
         return _get_object_or_404(queryset, *filter_args, **filter_kwargs)
-    except (TypeError, ValueError):
+    except (TypeError, ValueError, ValidationError):
```

Comment:
Shouldn't it re-raise it as a DRF validation error?

Reply:
The method is `get_object_or_404`, the `raise Http404` here is to keep it consistent with the `get_object_or_404` method that is implemented within the Django core.